exasol/nb_connector/extension_wrapper_common.py

```diff
@@ -53,30 +81,45 @@ def encapsulate_bucketfs_credentials(
     the connection object will instead turn the verification off. This is
     because there is no guarantee that the consumer of the connection object,
     i.e. a UDF, would have this custom CA list, and even if it would, its location
-    is unknown.
+    is unknown. This is only applicable for an On-Prem backend.
     """
 
-    bfs_host = conf.get(CKey.bfs_host_name, conf.get(CKey.db_host_name))
-    bfs_protocol = "https" if str_to_bool(conf, CKey.bfs_encryption, True) else "http"
-    bfs_dest = (
-        f"{bfs_protocol}://{bfs_host}:{conf.get(CKey.bfs_port)}/"
-        f"{conf.get(CKey.bfs_bucket)}/{path_in_bucket};{conf.get(CKey.bfs_service)}"
-    )
-    # TLS certificate verification option shall be provided in the fragment field.
-    verify: Optional[bool] = (False if conf.get(CKey.trusted_ca)
-                              else optional_str_to_bool(conf.get(CKey.cert_vld)))
-    if verify is not None:
-        bfs_dest += f'#{verify}'
+    backend = (bfs.path.StorageBackend.saas if get_backend(conf) == StorageBackend.saas
```

Comment:
The code below only uses the property `name` of variable `backend`.

Would it be better to base the if statement on the root condition as returned by `get_backend(conf)` and use a local function to get the name?

```python
def bfs_backend(backend):
    if backend == StorageBackend.saas:
        return bfs.path.StorageBackend.saas
    else:
        return bfs.path.StorageBackend.onprem

backend = get_backend(conf)
if backend == StorageBackend.saas:
    ... bfs_backend(backend).name
else: 
    ... bfs_backend(backend).name
```

Reply:
I don't think we need this `bfs_backend` function, but I take the point about the root condition.